File: 792-binary-search/binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
// Binary Search
// Time: O(log(n))
// Space: O(1)
var search = function(nums, target) {
    let low = 0, high = nums.length -1
    while (low < high) {
        let m = Math.floor((high + low) / 2) + 1;
        if (nums[m] === target) {
            return m
        } else if (target < nums[m]) {
            high = m - 1;
        } else {
            low = m;
        }
    }
    return nums[low] === target ? low : -1
};

File: 792-binary-search/binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int low = 0; int high = nums.length-1;

        while (low <= high) {
            int m = ((int) Math.floor((high+low) / 2) );
      

            if (nums[m] == target)             {
                return m;
            } else if (target < nums[m]) {
                high = m -1;
            } else {
                low = m+1;
            }
        }

        return -1;
    }
}